petab/core.py

```diff
@@ -741,6 +749,128 @@ def append_overrides(overrides):
     return df
 
 
+def get_priors_from_df(parameter_df: pd.DataFrame):
+    """Create list with information about the parameter priors
+
+    Arguments:
+        parameter_df: @type pandas.DataFrame
+    """
+
+    # get types and parameters of priors from dataframe
+    par_to_estimate = parameter_df.loc[parameter_df['estimate'] == 1]
+
+    prior_list = []
+    for _, row in par_to_estimate.iterrows():
+        # retrieve info about type
+        prior_type = str(row['priorType'])
+
+        # retrieve info about parameters of priors, make it a tuple of floats
+        tmp_pars = str(row['priorParameters']).split(';')
+        prior_pars = tuple([float(entry) for entry in tmp_pars])
+
+        # add parameter scale and bounds, as this may be needed
+        par_scale = row['parameterScale']
+        par_bounds = (row['lowerBound'], row['upperBound'])
+
+        # if no prior is specified, we assume a non-informative (uniform) one
+        if prior_type == 'nan':
+            prior_type = 'parameterScaleUniform'
+            prior_pars = (row['lowerBound'], row['upperBound'])
+
+        prior_list.append((prior_type, prior_pars, par_scale, par_bounds))
+
+    return prior_list
+
+
+def sample_from_prior(prior: tuple,
+                      n_starts: int):
+    """Creates samples based on prior
+
+    Arguments:
+        prior: @type tuple
+        n_starts: @type int
+    """
+
+    # unpack info
+    p_type, p_params, scaling, bounds = prior
+
+    # define a function to rescale the sampled points to parameter scale
+    def scale(x):
+        if scaling == 'lin':
+            return x
+        elif scaling == 'log':
+            return np.log(x)
+        elif scaling == 'log10':
+            return np.log10(x)
+        else:
+            raise NotImplementedError('Parameter priors on the parameter '
+                                      'scale ' + scaling + ' are currently '
+                                      'not implemented.')
+
+    def clip_to_bounds(x):
+        tmp_x = [min([bounds[1], ix]) for ix in x]
+        tmp_x = [max([bounds[0], ix]) for ix in tmp_x]
+        return np.array(tmp_x)
+
+    # define lambda functions for each parameter
+    if p_type == 'uniform':
+        sp = scale((p_params[1] - p_params[0]) * np.random.random((
+             n_starts,)) + p_params[0])
+
+    elif p_type == 'parameterScaleUniform':
+        sp = (p_params[1] - p_params[0]) * np.random.random((n_starts,
+                                                             )) + p_params[0]
+
+    elif p_type == 'normal':
+        sp = scale(np.random.normal(loc=p_params[0], scale=p_params[1],
+                                    size=(n_starts,)))
+
+    elif p_type == 'logNormal':
```

Comment:
We need to decide on a consistent parametrization of the distributions. 
I understood from our discussion (might be my misunderstanding...), that we agreed on another parametrization, but I prefer your parametrization here. What do you think @LeonardSchmiester? :)

Reply:
I thought that was the way we agreed on it: Introducing log-normal as an exponentiated normal distribution on the linear parameter space? What should be different?

Reply:
I thought that we parametrized it as it is parametrized in scipy. But as I said, I prefer your version and just wanted to make sure, that Leonard implements the same parametrization. I will adapt my code accordingly.

Reply:
Ah...
Yes, scipy does it differently...
Yes, but we do everything via numpy here, so I would also prefer not to get involved with yet another package which has redundant functionality, but uses a different definition...